packages/cozy-client/src/queries/dsl.js

```diff
@@ -150,10 +150,13 @@ class QueryDefinition {
    * @param {MangoSelector} selector   The Mango selector.
    * @returns {QueryDefinition}  The QueryDefinition object.
    */
-  where(selector) {
+  where(selector, mergeWithExisting) {
     this.checkSortOrder({ selector })
     this.checkSelector(selector)
-    return new QueryDefinition({ ...this.toDefinition(), selector })
+    return new QueryDefinition({
+      ...this.toDefinition(),
+      selector: (mergeWithExisting ? Object.assign({}, this.selector, selector) : selector)
```

Comment:
I think I need more context on that one and how you consider to use it

Reply:
You have a query  that takes care of the relationship, of the default sort, limit by,  and of a default filtering condition. You want to add a filtering condition to the already existing filters.

```
const todoQuery = Q('io.cozy.bank.operations').where({ done: false })
const recentTodoQuery = todoQuery.where({ date: { $gt: '2021-05' } }, true)
```

Without the `true`, we lose the { done: false } selector.

I think it'd be clearer with an options argument { merge: true }.

Reply:
Remembering my young years with Doctrine ORM, it seems to me that there was the possibility to add selectors later on thanks to `andWhere` method. 

```jsx
const todoQuery = Q('io.cozy.bank.operations').where({ done: false })
const recentTodoQuery = todoQuery.andWhere({ date: { $gt: '2021-05' } })
```

What do you think? 

Other questions in mind: 
- what about the indexesFields used? Imagine : 

```
const todoQuery = Q('io.cozy.bank.operations').where({ done: false }).indexFields(['done'])
const recentTodoQuery = todoQuery.where({ date: { $gt: '2021-05' } }, true) // or todoQuery.addWhere({ate: { $gt: '2021-05' } }
```

How can we add the new `date` index fields? Are we going to add this merge/add* everywhere?

Reply:
Yes maybe this pattern can be used for other methods of the DSL but

* for indexes I am not sure it is really easy as you have to control precisely the index does and it can be easier to just override the existing indexes.
 * For sort I think it the same story = easier to override than to add
 * limitBy, there is no sense in merging
 * getById, no sense in merging

For andWhere vs where(..., { merge: true }), I do not think there is really a difference, the main one for me would be that it seems that the discoverability of the andWhere vs the where is lower : at some point by typing the where method in your IDE, you'll be able to discover that there is the merge feature, whereas for andWhere, I find that it adds another method. The difference is not that great though, I do not have any a big leaning against one or another. It seems the merge method is a bit more flexible, at some point you could use it to implement a more granular merge strategy by passing a function.

.where(..., { merge: existingWhere => /* do what you want */ })

Anyway, andWhere could also support this via an optional function..

What do you think, do you have any preference ?

Reply:
If we plan to eventually allow the merge of index/sort/... , I have a slight preference for `mergeWhere`, and eventually `mergeIndex`, etc. The "merge" terminology seems a bit more intuitive to me :)
If we do not plan this, `.where(..., { merge: true}` looks fine to me. 
This being said, the merging of index and sort might be a bit risky, as the order is quite important and might be harder to see by such method.

Reply:
Finally, I removed the commit introducing this option. In Banks I ended up copying the selector because when I overrided the sort + indexFields, I had warnings because of the way the checks is done.

```
const aQuery = Q('io.cozy.bank.operations').where({ done: true })
  .indexFields(['done'])
  .sortBy([{ done: 'asc' }])

const bQuery = aQuery
	.where({ done: true, date: { $gt: '2020-01' } })
    .indexFields(['date', 'done']) // warning since data and srot fields are not the same
    .sortBy([{ date: 'asc'}, { done: 'asc'}]) // at this point the selector is OK
```

If the indexFields and sortBy calls are reversed, we still have a warning since the sortBy would tell us that we need to index the fields that we are sorting.

For the future, I am wondering if the query should only be validated at execution time. cc @paultranvan